### PR TITLE
Fix positive / negative prompts

### LINKED ISSUE #779: Stable Diffusion: providing a list of positive prompts and a list of negative prompts does not work as expected
### Describe the bug

See this forum post: https://discuss.huggingface.co/t/stable-diffusion-bs-1-uses-negative-as-prompt/24130.

In short:

```Python
_ = pipe(["frog"]*2, negative_prompt=["bird"]*2)
```

Reaches this condition https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L247. As expected, `negative_prompt` is a list with the same cardinality of `prompt` (otherwise an exception would have been raised). Therefore, in this case `uncond_input.input_ids` would have shape `(2, 77, 768)`. But then https://github.com/huggingface/diffusers/blob/main/src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py#L247 makes it become `(4, 77, 768)` because `batch_size` is 2. Therefore the concatenation operation in the next line creates `text_embeddings` with shape `(6, 77, 768)`. The first 4 ones are the ones corresponding to the negative prompt, and that's what gets passed to the model because the `latents` are correctly computed with shape `(2, 4, 64, 64)` (later expanded to batch size `4` during cfg).

### Reproduction

I could reproduce as explained above.

### Logs

_No response_

### System Info

diffusers @ main (`f3983d16eed57e46742d217363d8913bef7f748d`)

### PR DESCRIPTION
**Pull request by pcuenca**
I tested a few different combinations:
- Lists are supplied for both.
- Lists + num_images_per_prompt.
- Single positive, single negative, num_images_per_prompt > 1
- Same without negative prompts.

It seems to work, but it'd be nice to come up with a unit test somehow. Not easy to do because the code to be tested is inside the pipeline function.

Fixes #779.

### CODE REVIEW

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py [hunk 1]
```diff
@@ -283,7 +283,7 @@ def __call__(
 
             # duplicate unconditional embeddings for each generation per prompt, using mps friendly method
             seq_len = uncond_embeddings.shape[1]
-            uncond_embeddings = uncond_embeddings.repeat(batch_size, num_images_per_prompt, 1)
```
Comment (patil-suraj) [inline src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py:286]: The change looks good to me, but I'm surprised that this doesn't work, what was the issue with it ?

**Review by patil-suraj — APPROVED** [review]
Looks good to me, just posted one question in the comments. Also for tests ,maybe we could test each combination with dummy model

- single prompt
- list of prompts
- single prompt with `num_images_per_prompt`
- list of prompts with `num_images_per_prompt`

and the same including the negative prompt. This will add a lot of tests but I think it should be fine.

File: src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py (hunk 1 above)
Comment (pcuenca) [reply src/diffusers/pipelines/stable_diffusion/pipeline_stable_diffusion.py:286]: If we have three prompts and three negative prompts, batch_size would be three and we would be creating `uncond_embeddings` with shape `(9, 77, 768)` in this step. But `text_embeddings` is still `(3, 77, 768)`, and `latents_shape` in line 299 would be wrong.

**Comment by pcuenca** [comment]
I think we should merge this soon as more people are finding the problem (it used to be a silent error, but it now fails). See for example #1035. @patrickvonplaten @anton-l can one of you please review on Monday? Meanwhile I'll prepare a couple of tests.

**Review by patrickvonplaten — APPROVED** [review]

**Comment by pcuenca** [comment]
Superseded by #1035.

**Pull request closed without merging**